tests/compute/test_sampling.py

```diff
@@ -159,26 +232,49 @@ def _test_sampler(g, sampler, ntype):
     g = dgl.heterograph({
         ('item', 'bought-by', 'user'): ([0, 0, 1, 1, 2, 2, 3, 3], [0, 1, 0, 1, 2, 3, 2, 3]),
         ('user', 'bought', 'item'): ([0, 1, 0, 1, 2, 3, 2, 3], [0, 0, 1, 1, 2, 2, 3, 3])})
-    g = g.to(F.ctx())
+    if F.ctx() != F.cpu():
+        if use_uva:
+            g.create_formats_()
+            g.pin_memory_()
+        else:
+            g = g.to(F.ctx())
     sampler = dgl.sampling.PinSAGESampler(g, 'item', 'user', 4, 0.5, 3, 2)
     _test_sampler(g, sampler, 'item')
     sampler = dgl.sampling.RandomWalkNeighborSampler(g, 4, 0.5, 3, 2, ['bought-by', 'bought'])
     _test_sampler(g, sampler, 'item')
     sampler = dgl.sampling.RandomWalkNeighborSampler(g, 4, 0.5, 3, 2,
         [('item', 'bought-by', 'user'), ('user', 'bought', 'item')])
     _test_sampler(g, sampler, 'item')
+    if g.is_pinned():
+        g.unpin_memory_()
```

Comment:
Should we be unpinning the memory in a try/finally block to ensure if a test fails, we still unpin the graph?:
```
try:
  ...
finally:
  if g.is_pinned():
    g.unpin_memory()
```